src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,225 @@
+ * and limitations under the License.
+ */
+
+package com.cloudant.client.api;
+
+import static com.cloudant.client.org.lightcouch.internal.CouchDbUtil.assertNotEmpty;
+import static com.cloudant.client.org.lightcouch.internal.URIBuilder.buildUri;
+
+import com.cloudant.client.api.model.DesignDocument;
+import com.cloudant.client.org.lightcouch.CouchDatabase;
+import com.cloudant.client.org.lightcouch.CouchDatabaseBase;
+import com.cloudant.client.org.lightcouch.NoDocumentException;
+import com.cloudant.client.org.lightcouch.Response;
+import com.cloudant.client.org.lightcouch.internal.CouchDbUtil;
+import com.google.gson.Gson;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Provides API to work with design documents.
+ * <h3>Usage Example:</h3>
+ * <pre>
+ * // read from javascript design document file
+ * {@code
+ * DesignDocument design = DesignDocumentManager.fromFile("../example.js");
+ * }
+ *
+ * // read from directory of design documents
+ * {@code
+ * List<DesignDocument>DesignDocument design =
+ *                  DesignDocumentManager.fromDirectory("../design-files/");
+ * }
+ *
+ * // sync with the database
+ * {@code
+ * db.getDesignDocumentManager().put(designDoc);
+ * }
+ *
+ * // sync multiple design documents with the database
+ * {@code
+ * db.getDesignDocumentManager().put(listDesignDocuments);
+ * }
+ *
+ * // read from the database
+ * {@code
+ * DesignDocument designDoc2 = db.getDesignDocumentManager().get("_design/example");
+ * }
+ * </pre>
+ *
+ * @see {@link Database#getDesignDocumentManager()} to access the API.
+ * @see DesignDocument
+ * @since 2.0.0
+ */
+public class DesignDocumentManager {
+
+    private URI dbUri;
+    private CouchDatabaseBase dbc;
+    private CloudantClient client;
+
+    DesignDocumentManager(CouchDatabase db, CloudantClient client) {
+        this.dbUri = db.getDBUri();
+        this.dbc = db;
+        this.client = client;
+    }
+
+
+    /**
+     * Synchronizes a design document to the Database.
+     * <p>This method will first try to find a document in the database with the same id
+     * as the given document, if it is not found then the given document will be saved to the
+     * database.
+     * <p>If the document was found in the database, it will be compared with the given document
+     * using
+     * {@code equals()}. If both documents are not equal, then the given document will be saved to
+     * the
+     * database and updates the existing document.
+     *
+     * @param document The design document to synchronize
+     * @return {@link Response} as a result of a document save or update, or returns {@code null}
+     * if no
+     * action was taken and the document in the database is up-to-date with the given document.
+     */

```

Comment:
It looks like this is a copy from the class that you are replacing but the comment itself looks chopped up.  For example, lines 94, 95 should be on the same line.  Same sort of thing for 92,93 and 99,100.  Pretty minor stuff in the grand scheme of things.